calculation/seistech_calc/seistech_calc/plots/plt_plotting.py

```diff
@@ -549,3 +618,479 @@ def plot_uhs_branches(
+            bounds_y_range,
+            color="red",
+            linestyle="dotted",
+        )
+
+    if metadata == "sf":
+        plt.plot([1, 1], bounds_y_range, color="red", label="Reference point")
+
+    elif metadata == "vs30":
+        plt.plot(
+            [bounds.get(metadata).get("vs30"), bounds.get(metadata).get("vs30")],
+            bounds_y_range,
+            color="red",
+            label="Site-Specific $V_{s30}$",
+        )
+
+    plt.title(f"{CAUSAL_PARAMS_LABEL[metadata]} distribution")
+    plt.xlabel(f"{CAUSAL_PARAMS_LABEL[metadata]}")
+    plt.ylabel("Cumulative Probability, CDF")
+    plt.ylim(ymin=0)
+    plt.legend()
+
+    if save_file is not None:
+        plt.savefig(save_file)
+        plt.close()
+    else:
+        plt.show()
+
+
+def plot_gms_spectra(
+    gms_result: gms.GMSResult,
+    save_file: Path = None,
+):
+    """Plot of the pSA values of the realisations and
+     selected ground motions and the median, 16th,
+     and 84th percentile of the GCIM
+
+    Parameters
+    ----------
+    gms_result: gms.GMSResult
+    save_file: Path, optional
+    """
+    (
+        gcim_df,
+        realisations_df,
+        selected_gms_df,
+    ) = utils.calculate_gms_spectra(gms_result)
+
+    plt.figure(figsize=(20, 9))
+
+    for label, cur_gcim in gcim_df.iloc[:, 1:].iterrows():
+        plt.plot(
+            cur_gcim.index,
+            cur_gcim.values,
+            color="red",
+            linestyle="solid" if label == "median" else "dashdot",
+            label=GCIM_LABEL[label],
+        )
+
+    for cur_ix, cur_rel in realisations_df.iloc[:, 1:].iterrows():
+        plt.plot(
+            cur_rel.index.values,
+            cur_rel.values,
+            color="blue",
+            linestyle="solid",
+            label="Realisations" if cur_ix == 0 else None,
+            linewidth=0.4,
+        )
+
+    for cur_ix, cur_rel in selected_gms_df.iloc[:, 1:].iterrows():
+        plt.plot(
+            cur_rel.index.values,
+            cur_rel.values,
+            color="black",
+            linestyle="solid",
+            label="Selected Ground Motions" if cur_ix == 0 else None,
+            linewidth=0.4,
+        )
+
+    plt.xscale("log")
+    plt.yscale("log")
+    plt.xlabel("Period, T (s)")
+    plt.ylabel("Spectral acceleration, SA (g)")
+    plt.title("Pseudo acceleration response spectra")
+    plt.legend()
+
+    if save_file is not None:
+        plt.savefig(save_file)
+        plt.close()
+    else:
+        plt.show()
+
+
+def plot_gms_disagg_distribution(
+    contribution: List,
+    distribution: List,
+    gms_metadata: Dict,
+    metadata: str,
+    bounds: Dict = None,
+    save_file: Path = None,
+):
+    """CDF plots for the selected GMs and
+    all ruptures (disaggregation) for Magnitude
+    and Distance (Rrup)
+
+    Also shows causal bounding parameters if specified
+
+    Parameters
+    ----------
+    contribution: List
```

Comment:
Both the `contribution` and `distribution` parameter are unclear as to what they are?
Please specify where they come from, or preferably use the result objects, i.e. GMSResult and DisaggResult

Reply:
Also whats the difference between `gms_metadata` and `metadata`

Reply:
Added some extra explanation for `contribution` and `distribution`.
Also instead of using `gms_metadata`, accepting gmsResult object with `metadata_key` instead of `metadata`